app/models/manageiq/providers/azure/refresh_helper_methods.rb

```diff
@@ -116,56 +116,83 @@ def network_routers
   end
 
   # Create the necessary service classes and lock down their api-version
-  # strings using the config/settings.yml from the provider repo. The
-  # "to_s" call for the version strings puts the date in the format
-  # that we need, i.e. "YYYY-MM-DD".
+  # strings using the config/settings.yml from the provider repo.
+
+  def cached_resource_provider_service(config)
+    @cached_resource_provider_service ||= resource_provider_service(config)
```

Comment:
if this is something that rarely changes, we could probably wrap it to cache_with_timeout block?

Reply:
I didn't know about `cache_with_timeout` but perhaps we can switch it to that if you prefer.

Reply:
yeah, for the things that do not change (or not frequently) it's perfect